### PR TITLE
Remove plugin API & split project into crates

### PR DESCRIPTION
**Pull request by bkolobara**
This PR does 2 changes:

### 1. Removes the plugin subsystem + associated host function APIs

In the current shape the plugin system is undocumented and not particularly useful. It has a lot of code that touches many parts of the codebase. We don't even have a good use case for it at the moment and this PR removes is just to simplify the rest of our codebase.

The following issue tracks some ideas around a more useful plugin system: #69.

### 2. Splits some host functions into separate crate:

- [x] Error handling
- [x] Messaging
- [x] Networking
- [x] Version
- [ ] Process/Module/Environment
- [x] WASI

This change will allow us to feature gate some of the dependencies if lunatic is used in an embedded context and not all host functions are required. It is also the first step in abstracting the process state into something more generic that can be extended.

This required some other parts of the runtime to be separated out into crates too, to avoid circular dependencies. The host functions that deal with process spawning, modules and environments are not separated out. They are too entangled with the `Node` struct (distributed) and can't be pulled out at the moment (@withtypes).

WASI could also be separated out at one point.

### CODE REVIEW

**Comment by sleipnir** [comment]
> This change will allow us to feature gate some of the dependencies if lunatic is used in an embedded context

This is likely to be particularly useful in integrating with Krustlet, if there is a way to run the lunatic runtime programmatically

**Comment by bkolobara** [comment]
@sleipnir I'm very much interested in bringing lunatic support to Krustlet. It should be even possible at this stage to turn lunatic into a Krustlet provider. 

Lunatic has 2 features that other providers don't offer:
1. A concurrency model based on processes
2. Networking primitives

This should allow people to already deploy some useful applications, like web apps, to k8s. Until now I just saw simple demo apps running on Krustlet. I don't have much time to look into it at the moment, but could assist if someone wants to take onto the task. Looking at some of the other providers could be a good starting point, like https://github.com/krustlet/krustlet/pull/677.

**Comment by sleipnir** [comment]
Yes I have been following and researching how this would be implemented for a long time, I hope to soon be able to provide a proper implementation for Lunatic

**Review by withtypes — APPROVED** [review]
I like the idea and the code seems ok. I've tested it with my toy distributed ping-pong example. I can try and move wasi to a separate crate...

File: src/message.rs [hunk 1]
```diff
@@ -1,194 +1 @@
-/*!
```
Comment (withtypes) [inline src/message.rs:1]: I noticed this wasn't deleted...

**Pull request merged**